datahub-web-react/src/app/shared/tags/TagTermGroup.tsx

```diff
@@ -173,14 +177,28 @@ export default function TagTermGroup({
             {domain && (
                 <DomainLink urn={domain.urn} name={entityRegistry.getDisplayName(EntityType.Domain, domain) || ''} />
             )}
-            {uneditableGlossaryTerms?.terms?.map((term) => (
-                <TermLink to={entityRegistry.getEntityUrl(EntityType.GlossaryTerm, term.term.urn)} key={term.term.urn}>
-                    <Tag closable={false}>
-                        <BookOutlined style={{ marginRight: '3%' }} />
-                        {entityRegistry.getDisplayName(EntityType.GlossaryTerm, term.term)}
-                    </Tag>
-                </TermLink>
-            ))}
+            {uneditableGlossaryTerms?.terms?.map((term) => {
+                renderedTags += 1;
+                if (maxShow && renderedTags > maxShow)
+                    return (
+                        <TagText>
+                            {uneditableGlossaryTerms?.terms
+                                ? `+${uneditableGlossaryTerms?.terms?.length - maxShow}`
+                                : null}
+                        </TagText>
+                    );
```

Comment:
I think we have a UI bug here - if `maxShow` is 3 and there are 5 `uneditableGlossaryTerms`, we would then show 3 tags, then "+2" twice (instead of just once like we would want)

the easiest way to fix this may be to have another logic check, so overall the logic could look like:
```
if (maxShow && renderedTags === maxShow + 1) // return + the number remaining
else if (maxShow && renderedTags > maxShow) return null;
// return the normal <TermLink> wrapped <Tag> below
```

Reply:
Agree top your point @chriscollins3456. Thanks for highlighting, the issue.  
The new changes are added in the same PR.